test/python/primitives/test_backend_estimator_dev.py

```diff
@@ -0,0 +1,924 @@
+        - target_qubits: the number of qubits to target during transpilation
+        - layout_intlist: indices to map virtual qubits to during transpilaiton
+        - measured_qubits: virtual qubits to measure
+
+    Returns:
+        - Transpiled base circuit: with a `final_layout` entry in its metadata
+        - Measurement circuit: before transpilation (i.e. no layout applied)
+        - Transpiled measurement circuit: with `final_layout` applied
+    """
+    _, transpiled_base = build_base_circuit(target_qubits, layout_intlist)
+    transpiled_base.metadata.update({"extra_base": Mock()})
+    measurement, transpiled_measurement = build_measurement_circuit(
+        target_qubits, layout_intlist, measured_qubits
+    )
+    measurement.metadata.update({"extra_meas": Mock()})
+    return transpiled_base, measurement, transpiled_measurement
+
+
+def build_base_circuit(target_qubits, layout_intlist):
+    """Build example base and transpiled base circuits."""
+    num_qubits = len(layout_intlist)
+    base = QuantumCircuit(num_qubits)
+    base.h(range(num_qubits))  # Dummy single-qubit gates
+    if num_qubits > 1:
+        base.cx(range(-1, num_qubits - 1), range(num_qubits))  # Dummy two-qubit gates
+    transpiled_base = transpile_to_layout(base, target_qubits, layout_intlist)
+    return base, transpiled_base
+
+
+def build_measurement_circuit(target_qubits, layout_intlist, measured_qubits):
+    """Build example measurement and transpiled measurement circuits."""
+    num_qubits = len(layout_intlist)
+    num_measurements = len(measured_qubits)
+    measurement = QuantumCircuit(num_qubits, num_measurements)
+    measurement.h(measured_qubits)  # Dummy gates (i.e. X measurements)
+    measurement.measure(measured_qubits, range(num_measurements))
+    measurement.metadata = {"measured_qubit_indices": measured_qubits}
+    transpiled_measurement = transpile_to_layout(measurement, target_qubits, layout_intlist)
+    return measurement, transpiled_measurement
+
+
+def transpile_to_layout(circuit, target_qubits, layout_intlist):
+    """Transpile circuit to match a given layout intlist."""
+    if circuit.num_qubits != len(layout_intlist):
+        raise ValueError("Circuit incompatible with requested layout.")
+    if circuit.num_qubits > target_qubits:
+        raise ValueError("Circuit incompatible with requested target.")
+    layout_dict = dict.fromkeys(range(target_qubits))
+    layout_dict.update(dict(zip(layout_intlist, circuit.qubits)))
+    applied_layout = Layout(layout_dict)
+    passes = [SetLayout(layout=applied_layout), ApplyLayout()]
+    pass_manager = PassManager(passes=passes)
+    transpiled = pass_manager.run(circuit)
+    transpiled.metadata = {"final_layout": applied_layout}
+    return transpiled
+
+
+################################################################################
+## UNIT TESTS
+################################################################################
+@ddt
+class TestTranspilation(TestCase):
```

Comment:
Qiskit unittests need to be subclasses of `qiskit.test.QiskitTestCase`.

Reply:
My bad, thanks, fixed!